tests/drivers/flash_simulator/flash_sim_impl/src/main.c

```diff
@@ -325,12 +325,14 @@ ZTEST(flash_sim_api, test_double_write)
 	rc = flash_write(flash_dev, FLASH_SIMULATOR_BASE_OFFSET,
 				 &data, sizeof(data));
 	zassert_equal(-EIO, rc, "Unexpected error code (%d)", rc);
-}
+#else
+	ztest_test_skip();
 #endif
+}
 
-#if !defined(CONFIG_FLASH_SIMULATOR_EXPLICIT_ERASE)
 ZTEST(flash_sim_api, test_ramlike)
 {
+#if defined(CONFIG_FLASH_SIMULATOR_RAMLIKE)
```

Comment:
Similarly like above - as long as it would compile (for example option does not remove some structure fields from existence) i would opt for compilation level conditionals.